openslides_backend/action/motion_category/number_motions.py

```diff
@@ -0,0 +1,225 @@
+                lead_motion_id = self.get_lead_motion_id(motion_id)
+                if lead_motion_id and lead_motion_id not in affected_motions:
+                    raise ActionException(
+                        f'Amendment "{motion_id}" cannot be numbered, because it\'s lead motion ({lead_motion_id}) is not in category {instance["id"]} or any subcategory.'
+                    )
+
+            # generate number_value_map with the number_values for motion_ids.
+            main_counter = 1
+            lead_motions_counter: Dict[int, int] = dict()
+            number_value_map: Dict[int, int] = dict()
+            for motion_id in affected_motions:
+                lead = self.get_lead_motion_id(motion_id)
+                if lead:
+                    if lead in lead_motions_counter:
+                        number_value_map[motion_id] = lead_motions_counter[lead]
+                        lead_motions_counter[lead] += 1
+                    else:
+                        number_value_map[motion_id] = 1
+                        lead_motions_counter[lead] = 2
+                else:
+                    number_value_map[motion_id] = main_counter
+                    main_counter += 1
+
+            for motion_id in affected_motions:
+                number = self.get_number(motion_id, number_value_map)
+                if number in existing_numbers:
+                    raise ActionException(
+                        f'Numbering aborted because the motion identifier "{number}" already exists.'
+                    )
+
+                yield {
+                    "id": motion_id,
+                    "number": number,
+                }
+
+    def init_memory(self, main_category_id: int) -> None:
+        """Preload all categories with needed fields, all motions with needed fields
+        and meeting with needed fields."""
+        category = self.datastore.get(
+            FullQualifiedId(Collection("motion_category"), main_category_id),
+            ["meeting_id"],
+        )
+
+        if category.get("meeting_id"):
+            meeting = self.datastore.get(
+                FullQualifiedId(Collection("meeting"), category["meeting_id"]),
+                ["motion_ids", "motion_category_ids"],
+            )
+        else:
+            meeting = {}
+
+        gmr_categories = GetManyRequest(
+            Collection("motion_category"),
+            meeting.get("motion_category_ids", []),
+            ["prefix", "parent_id", "child_ids", "weight", "motion_ids"],
+        )
+        gmr_motions = GetManyRequest(
+            Collection("motion"),
+            meeting.get("motion_ids", []),
+            [
+                "lead_motion_id",
+                "meeting_id",
+                "category_weight",
+                "category_id",
+                "number",
+            ],
+        )
+        gmr_meetings = GetManyRequest(
+            Collection("meeting"),
+            [category.get("meeting_id", 1)],
```

Comment:
This is solved with the above comment, but you cannot just assume a meeting id, this would lead to potential security flaws!

Reply:
Removed this gmr_meetings completely.